src/client/client.js

```diff
@@ -175,16 +150,22 @@ class Client {
     return `Basic ${encoded}`;
   }
 
-  sendRequest(options) {
-    return new Promise((resolve, reject) => {
-      this._request(options, (error, response, result) => {
-        if (error) {
-          reject(error);
-        } else {
-          resolve({response, result});
-        }
-      });
-    });
+  fetchWithOptions(uri, options) {
+    return fetch(options.uri, options);
+  }
+
+  async sendRequest(options) {
+    const response = await this.fetchWithOptions(options.uri, options);
+
+    let result = {};
+    if (
+      response.status !== 204 &&
+      response.headers.get('content-type')?.includes('application/json')
+    ) {
+      result = await response.json();
+    }
```

Comment:
@teebot I modified your code here, successful delete from zendesk returns null and the original way we treated this is to just pass that along back to the client in the result with a soft non throwing error. I added a check to ensure that the content type in the response is json before we call `response.json()` (Note: we could instead call `response.text()`, but I am not aware that the zendesk api actually returns text, even with errors it's returnign json... only the delete api call is the one strange one)